CHANGELOG.md

```diff
@@ -6,6 +6,93 @@
+
+Developers can now provide editor extensions, like themes and programming language parsers, through new application registries.
+
+### New extension manager
+
+Starting with JupyterLab 3, extensions can be installed via Python packages.
+
+In JupyterLab 4, building on this feature, the Extension Manager can now list and install extensions from [pypi.org](https://pypi.org/search/?c=Framework+%3A%3A+Jupyter+%3A%3A+JupyterLab).
+
+Developers can provide an alternative package repository to display their own set of extensions.
+
+### Improved document search
+
+The Search and Replace functionality has been improved with new features when searching in a notebook:
+
+- Highlight matches in rendered markdown cells
+- Search in selection
+- Multi-line search
+- Replace using regex capture-group references
+- Replace while preserving case
+
+### UI improvements
+
+Some new elements have been added or changed in the UI:
+
+- Rework the running kernels section
+- "Add cell" button at the notebook bottom
+- Dialog to display keyboard shortcuts as in the Classic Notebook (use <kbd>CTRL+SHIFT+H</kbd>)
+- Display first line of cell input and outputs when they are collapsed
+
+### Accessibility improvements
+
+JupyterLab is not yet fully accessible. Currently the focus is on making Notebook 7 accessible.
+A big part of the code is shared, though, and the following accessibility improvements are in JupyterLab 4:
+
+- Improved focus and keyboard navigation in the file browser
+- Add more roles and labels to UI elements
+- Main menu collapses to a hamburger menu if the web page is not wide enough to display all items.
+
+### Performance enhancements
+
+JupyterLab is now faster, thanks to the following improvements:
+
+- CSS rules optimization: CSS selectors have been optimized to improve web browser performance when many elements are present on a page.
+- Upgrade to CodeMirror 6: Especially for notebooks with many cells, the new CodeMirror version is far more efficient than the previous version. Large notebook should now load faster.
+- Notebook windowing: By rendering only the parts of a notebook that fit in the web browser viewport, JupyterLab is much more efficient. See an important note below.
+
+_Notebook windowing_ might add side effects with notebooks containing cells injecting new CSS styles or JavaScript. It may also be less convenient if some cell outputs are displaying [iframes](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe). If you have issues with notebook rendering, try changing the user setting _Notebook_ > _Windowing mode_ to `defer` or `none`. (`none` should be used as a last resort, because it disables all optimizations.)
```

Comment:
```suggestion
Notebook windowing _might_ add side effects with notebooks containing cells injecting new CSS styles or JavaScript. It may also be less convenient if some cell outputs are displaying [iframes](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe). If you have issues with notebook rendering, try changing the user setting _Notebook_ > _Windowing mode_ to `defer` or `none`. (`none` should be used as a last resort, because it disables all optimizations.)
```